core/src/Kokkos_Macros.hpp

```diff
@@ -148,12 +149,14 @@
 #define KOKKOS_COMPILER_APPLECC __APPLE_CC__
 #endif
 
-#if defined(__clang__) && !defined(KOKKOS_COMPILER_INTEL)
+#if defined(__clang__) && !defined(KOKKOS_COMPILER_INTEL) && \
+    !defined(KOKKOS_COMPILER_INTEL_LLVM)
```

Comment:
```suggestion
#if defined(__clang__) && !defined(KOKKOS_COMPILER_INTEL_LLVM)
```
I don't think there's a case where `KOKKOS_COMPILER_INTEL` is relevant to this, but the `LLVM` macro isn't defined.